tests/test_mpls.py

```diff
@@ -0,0 +1,1049 @@
+import os
+import re
+import time
+import json
+import pytest
+
+from swsscommon import swsscommon
+from dvslib.dvs_common import wait_for_result
+
+class TestMplsBase(object):
+    def setup_db(self, dvs):
+        self.pdb = dvs.get_app_db()
+        self.adb = dvs.get_asic_db()
+        self.cdb = dvs.get_config_db()
+
+    def set_admin_status(self, interface, status):
+        self.cdb.update_entry("PORT", interface, {"admin_status": status})
+
+    def create_l3_intf(self, interface):
+        self.cdb.create_entry("INTERFACE", interface, {"NULL": "NULL"})
+
+    def remove_l3_intf(self, interface):
+        self.cdb.delete_entry("INTERFACE", interface)
+
+    def create_mpls_intf(self, interface):
+        self.cdb.create_entry("INTERFACE", interface, {"mpls": "enable"})
+
+    def remove_mpls_intf(self, interface):
+        self.cdb.delete_entry("INTERFACE", interface)
+
+    def add_ip_address(self, interface, ip):
+        self.cdb.create_entry("INTERFACE", interface + "|" + ip, {"NULL": "NULL"})
+
+    def remove_ip_address(self, interface, ip):
+        self.cdb.delete_entry("INTERFACE", interface + "|" + ip)
+
+    def create_route_entry(self, key, pairs):
+        tbl = swsscommon.ProducerStateTable(self.pdb.db_connection, "ROUTE_TABLE")
+        fvs = swsscommon.FieldValuePairs(list(pairs.items()))
+        tbl.set(key, fvs)
+
+    def remove_route_entry(self, key):
+        tbl = swsscommon.ProducerStateTable(self.pdb.db_connection, "ROUTE_TABLE")
+        tbl._del(key)
+
+    def check_route_entries(self, destinations):
+        def _access_function():
+            route_entries = self.adb.get_keys("ASIC_STATE:SAI_OBJECT_TYPE_ROUTE_ENTRY")
+            route_destinations = [json.loads(route_entry)["dest"]
+                                  for route_entry in route_entries]
+            return (all(destination in route_destinations for destination in destinations), None)
+
+        wait_for_result(_access_function)
+
+    def get_asic_db_key(self, destination):
```

Comment:
this is also not getting used. Can you remove unused APIs if not planning on using in this PR.